File: day6/two.fs
let findMarkerIndex (signal: string) (marker: int) =
    let rec loop i =
        let current = signal.[i..marker] |> Seq.distinct |> Seq.length
        if  current = marker
        then i + marker
        else (loop (i + 1))
      
    loop 0

printfn "%i" (findMarkerIndex "bvwbjplbgvbhsrlpgdmjqwftvncz" 14)

File: day6/two.py
def find_marker_index(signal, marker):
	for i in range(len(signal)):
		if len(set(signal[i:i+marker])) == marker:
			return i + marker

	return -1;

s = "bvwbjplbgvbhsrlpgdmjqwftvncz"
marker = 14
print(find_marker_index(s, marker))

